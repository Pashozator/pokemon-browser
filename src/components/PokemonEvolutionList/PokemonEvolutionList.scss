@import '../../styles/media-queries';

.pokemon-evolution-list {
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
  display: flex;

  @include for-up-sm() {
    align-items: center;
    flex-direction: row;
  }
}
