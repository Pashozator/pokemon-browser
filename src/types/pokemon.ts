export interface Pokemon {
	id: string;
	name: string;
	image: string;
	maxHP?: number;
	fleeRate?: number;
	maxCP?: number;
	classification?: string;
	types?: string[];
	resistant?: string[];
	weaknesses?: string[];
	evolutions?: Pokemon[];
}
