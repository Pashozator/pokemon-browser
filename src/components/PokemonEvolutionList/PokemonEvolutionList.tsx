import React from 'react';
import './PokemonEvolutionList.scss';
import { Pokemon } from '../../types/pokemon';
import PokemonEvolutionCard from '../PokemonEvolutionCard/PokemonEvolutionCard';

interface PokemonEvolutionsProps {
	evolutions: Pokemon[] | undefined;
}

const PokemonEvolutionList: React.FC<PokemonEvolutionsProps> = ({ evolutions = [] }) => {
	return (
		<div className="pokemon-evolution-list">
			{evolutions.map(evolution => <PokemonEvolutionCard key={evolution.id} evolution={evolution}/>)}
		</div>
	);
};

export default PokemonEvolutionList;
