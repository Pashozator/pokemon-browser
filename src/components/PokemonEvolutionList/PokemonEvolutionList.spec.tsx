import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Pokemon } from '../../types/pokemon';
import PokemonEvolutionList from './PokemonEvolutionList';

describe('PokemonEvolutionList component', () => {
	it('should render', () => {
		const evolutions: Pokemon[] = [{
			id: 'test',
			name: 'Pikachu',
			image: 'image'
		}];

		const wrapper = shallow(<PokemonEvolutionList evolutions={evolutions}/>);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
