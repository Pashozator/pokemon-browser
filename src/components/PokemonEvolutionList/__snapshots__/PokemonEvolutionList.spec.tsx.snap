// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonEvolutionList component should render 1`] = `
<div
  className="pokemon-evolution-list"
>
  <PokemonEvolutionCard
    evolution={
      Object {
        "id": "test",
        "image": "image",
        "name": "Pikachu",
      }
    }
    key="test"
  />
</div>
`;
